src/utils.py

```diff
@@ -1,296 +1,305 @@
+        with contextlib.suppress(Exception):
+            self.webdriver.find_element(By.ID, "cookie-banner").find_element(
+                By.TAG_NAME, "button"
+            ).click()
+            time.sleep(2)
+
+    def tryDismissBingCookieBanner(self):
+        with contextlib.suppress(Exception):
+            self.webdriver.find_element(By.ID, "bnp_btn_accept").click()
+            time.sleep(2)
+
+    def switchToNewTab(self, timeToWait: int = 0):
+        time.sleep(0.5)
+        self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
+        if timeToWait > 0:
+            time.sleep(timeToWait)
+
+    def closeCurrentTab(self):
+        self.webdriver.close()
+        time.sleep(0.5)
+        self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[0])
+        time.sleep(0.5)
+
+    def visitNewTab(self, timeToWait: int = 0):
+        self.switchToNewTab(timeToWait)
+        self.closeCurrentTab()
+
+    def getRemainingSearches(self) -> RemainingSearches:
+        dashboard = self.getDashboardData()
+        searchPoints = 1
+        counters = dashboard["userStatus"]["counters"]
+
+        progressDesktop = counters["pcSearch"][0]["pointProgress"]
+        targetDesktop = counters["pcSearch"][0]["pointProgressMax"]
+        if len(counters["pcSearch"]) >= 2:
+            progressDesktop = progressDesktop + counters["pcSearch"][1]["pointProgress"]
+            targetDesktop = targetDesktop + counters["pcSearch"][1]["pointProgressMax"]
+        if targetDesktop in [30, 90, 102]:
+            searchPoints = 3
+        elif targetDesktop == 50 or targetDesktop >= 170 or targetDesktop == 150:
+            searchPoints = 5
+        remainingDesktop = int((targetDesktop - progressDesktop) / searchPoints)
+        remainingMobile = 0
+        if dashboard["userStatus"]["levelInfo"]["activeLevel"] != "Level1":
+            progressMobile = counters["mobileSearch"][0]["pointProgress"]
+            targetMobile = counters["mobileSearch"][0]["pointProgressMax"]
+            remainingMobile = int((targetMobile - progressMobile) / searchPoints)
+        return RemainingSearches(desktop=remainingDesktop, mobile=remainingMobile)
+
+    @staticmethod
+    def formatNumber(number, num_decimals=2):
+        return pylocale.format_string(
+            f"%10.{num_decimals}f", number, grouping=True
+        ).strip()
+
+    @staticmethod
+    def getBrowserConfig(sessionPath: Path) -> dict:
+        configFile = sessionPath.joinpath("config.json")
+        if configFile.exists():
+            with open(configFile, "r") as f:
+                return json.load(f)
+        else:
+            return {}
+
+    @staticmethod
+    def saveBrowserConfig(sessionPath: Path, config: dict):
+        configFile = sessionPath.joinpath("config.json")
+        with open(configFile, "w") as f:
+            json.dump(config, f)
+
+    @staticmethod
+    def isDebuggerAttached() -> bool:
+        return sys.gettrace() is not None
```

Comment:
You need to import `sys` at the top of utils.py. Currently this errors out because it's not imported.

Reply:
Yea this issue is back again @cal4

Reply:
Got a few different changelists, tendency for this to happen. Got rid of the debugger attached thing. Caused more trouble than worth in other ways.